README.markdown

```diff
@@ -223,10 +223,13 @@ For UIKit view controllers, consider grouping lifecycle, custom accessors, and I
 
 ### Unused Code
 
-Unused (dead) code, including Xcode template code and placeholder comments should be removed.
+Xcode template code and placeholder comments should **not** be removed.
+
+Other unused (dead) code should be removed and should never reach `development` branch. You can keep commented code in your branch or even in a feature branch, but `development` branch should be clean from commented code. If a future feature needs some code that has been removed, it can always be retrieved using git. Mark your commit name accordingly when removing code if it is susceptible to be used in the future, it will make it easier to cherry pick.
 
```

Comment:
I hope we are all in agreement with this. I have come across a ton of dead code (in the editor, in merchant, in achievements...) and I think we need to start cleaning up our code. If we are implementing some new feature and come across some dead code in the same file, we should remove it.

Reply:
Not sure why we want to keep the Xcode template code and placeholder comments.

I mostly agree with the other stuff, but I think that we really want a format for commit messages where we're specifically removing something we think we want in the future. It is incredibly easy to give a commit message a descriptive name and then never be able to remember what you called a specific thing 6 months later when you need to find it in the git history. I think having some formula for the messages makes that searching way easier. Maybe something like `REMOVED: <thing you removed>` idk what the formula should be exactly.

Reply:
> Not sure why we want to keep the Xcode template code and placeholder comments.

About this, I agree. I used to have my Xcode set up without template code. On a previous company we had to submit our code to a due diligence and all comments had to be removed (and it was a pain). The reason I added this line of code is because currently is the way we are doing it. Aren't we?

About your other comment, I like the idea of finding a formula for naming specific commits that we may need in the future. I also used the approach of saving patches as files, but this is harder to keep in sync with the rest of the team.

Reply:
We are not intentionally keeping Xcode template code or placeholder comments. If they are there it is a mistake.

Reply:
I think we may be referring to different comments.

I guess we want to keep this:

```swift
//
//  ShippingRatesCollectionViewCell.swift
//  Universe
//
//  Created by Victor Baro Fernandez on 8/31/20.
//  Copyright © 2020 Future Lab, Inc. All rights reserved.
//
```

But remove all the rest, like:
```swift
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
```

Right? 
Or we also want to remove the header comment?

Reply:
We want to keep the header comment, but not the other stuff Xcode adds

Reply:
👍 I will update the readme accordingly

Reply:
Do we want a commit message formula for removing code or no?

Reply:
Oh yes, let's do it. I will add a comment about it. Shall we go with 
`REMOVED: <whatever>` ?

Reply:
I'm ok with that, but open to suggestions for something better

Reply:
Bumping this

Reply:
I think if a commit is only removing code, then a simple `Removed: <whatever>` is nice. And if you are considering removing code and making some other changes, I'd try to keep those commits separate. But both are these are highly circumstantial. I'm not opposed to them being suggested in the guidelines but our ability to enforce them effectively is pretty limited because there are lots of judgement calls for when to commit, the order of those commits determines how effectively something can be restored anyway, and if we do notice a problem and request a fix the act of fixing it is a big pain involving resetting multiple commits then carefully adding the changes you want to commit one at a time.

So suggestion, callout is all good. But don't think this needs much more than that, because it is a rare and also unenforceable issue.

Reply:
👍

Reply:
So we do want to add a suggested commit formula right?

Reply:
https://github.com/onuniverse/swift-style-guide/pull/6/commits/c38333749852853b60a546226edba1c31bbddb43